aptos-move/framework/aptos-stdlib/sources/cryptography/bulletproofs.move

```diff
@@ -0,0 +1,253 @@
+    // Constants
+    //
+
+    /// The maximum range supported by the Bulletproofs library is [0, 2^{64}).
+    const MAX_RANGE_BITS : u64 = 64;
+
+    //
+    // Error codes
+    //
+
+    /// There was an error deserializing the range proof.
+    const E_DESERIALIZE_RANGE_PROOF: u64 = 1;
+
+    /// The committed value given to the prover is too large.
+    const E_VALUE_OUTSIDE_RANGE: u64 = 2;
+
+    /// The range proof system only supports proving ranges of type $[0, 2^b)$ where $b \in \{8, 16, 32, 64\}$.
+    const E_RANGE_NOT_SUPPORTED: u64 = 3;
+
+    /// The native functions have not been rolled out yet.
+    const E_NATIVE_FUN_NOT_AVAILABLE: u64 = 4;
+
+    //
+    // Structs
+    //
+
+    /// Represents a zero-knowledge range proof that a value committed inside a Pedersen commitment lies in
+    /// `[0, 2^{MAX_RANGE_BITS})`.
+    struct RangeProof has copy, drop, store {
+        bytes: vector<u8>
+    }
+
+    //
+    // Public functions
+    //
+
+    /// Returns the maximum # of bits that the range proof system can verify proofs for.
+    public fun get_max_range_bits(): u64 {
+        MAX_RANGE_BITS
+    }
+
+    /// Deserializes a range proof from a sequence of bytes. The serialization format is the same as the format in
+    /// the zkcrypto's `bulletproofs` library (https://crates.io/crates/bulletproofs).
+    public fun range_proof_from_bytes(bytes: vector<u8>): RangeProof {
+        RangeProof {
+            bytes
+        }
+    }
+
+    /// Returns the byte-representation of a range proof.
+    public fun range_proof_to_bytes(proof: &RangeProof): vector<u8> {
+        proof.bytes
+    }
+
+    /// Verifies a zero-knowledge range proof that the value `v` committed in `com` (under the default Bulletproofs
+    /// commitment key; see `pedersen::new_commitment_for_bulletproof`) satisfies $v \in [0, 2^b)$. Only works
+    /// for $b \in \{8, 16, 32, 64\}$. Additionally, checks that the prover used `dst` as the domain-separation
+    /// tag (DST).
+    ///
+    /// WARNING: The DST check is VERY important for security as it prevents proofs computed for one application
+    /// (a.k.a., a _domain_) with `dst_1` from verifying in a different application with `dst_2 != dst_1`.
+    public fun verify_range_proof_pedersen(com: &pedersen::Commitment, proof: &RangeProof, num_bits: u64, dst: vector<u8>): bool {
+        assert!(features::bulletproofs_enabled(), error::invalid_state(E_NATIVE_FUN_NOT_AVAILABLE));
+
+        verify_range_proof_internal(
+            ristretto255::point_to_bytes(&pedersen::commitment_as_compressed_point(com)),
+            &ristretto255::basepoint(), &ristretto255::hash_to_point_base(),
+            proof.bytes,
+            num_bits,
+            dst
+        )
+    }
+
+    /// Verifies a zero-knowledge range proof that the value `v` committed in `com` (as v * val_base + r * rand_base,
+    /// for some randomness `r`) satisfies $v \in [0, 2^{num_bits})$. Only works for `num_bits` \in {8, 16, 32, 64}.
```

Comment:
Better to have a consistent style.
```suggestion
    /// for some randomness `r`) satisfies `v` in `[0, 2^num_bits)`. Only works for `num_bits` in `{8, 16, 32, 64}`.
```

Reply:
Oh, but we should use LaTeX as our consistent style, if we can, because it's rendered nicely in the documents! Sometimes, when the variable names are too long, LaTeX formatting looks weird, so we should not.